pkgs/applications/science/psychopy/default.nix

```diff
@@ -0,0 +1,92 @@
+{ lib
+, fetchurl
+, fetchgit
+, fetchhg
+, callPackage
+, python310Packages
+, fetchFromGitHub
+, fetchPypi
+, qt6
+, pkg-config
+, ffmpeg-full
+, glibc
+}:
+
+python310Packages.buildPythonApplication rec {
+  pname = "psychopy";
+  version = "2023.2.4";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "psychopy";
+    repo = "psychopy";
+    rev = version;
+    hash = "sha256-2sE7CauMe7TlZocBFM/ftsTpSxTifHUyHTqy1rupn9A=";
+  };
+
+  nativeBuildInputs = [
+    qt6.wrapQtAppsHook
+    pkg-config
+  ];
+  buildInputs = [
+  ];
+  propagatedBuildInputs = with python310Packages; [
+    pdm-backend
+    distro
+    tomlkit
+    setuptools
+    six
+    packaging
+
+    numpy
+    scipy
+    matplotlib
+    pyglet
+    pillow
+    pyqt6
+    pandas
+    #questplus
+    openpyxl
+    xmlschema
+
+    soundfile
+    imageio
+    imageio-ffmpeg
+    #psychtoolbox
+
+    gevent
+    psutil
+    pyzmq
+    ujson
+    msgpack
+    msgpack-numpy
+    freetype-py
+    python-bidi
+    arabic-reshaper
+    websockets
+    wxPython_4_2
+    markdown-it-py
+    requests
+    future
+    python-gitlab
+    gitpython
+    cryptography
+    astunparse
+    esprima
+    jedi
+    pyserial
+    opencv4
+    python-vlc
+    xlib
+    tables
+    javascripthon
+    pyparallel
+    ffpyplayer
+  ] ++ [
```

Comment:
this `] ++ [` is unnecessary.

Reply:
No, the first array has `with python310Packages;` in front of it.